docs/proposals/20231122-automate-aks-features.md

```diff
@@ -0,0 +1,360 @@
+
+It also has the drawback that one version of CAPZ is tied directly to a single AKS API version. The spec could
+potentially contain separate fields for each API version and enforce in webhooks that only one API version is
+being used at a time. Alternatively, users may set fields only present in a newer API version directly on the
+ManagedCluster after creation (if allowed by AKS) because CAPZ will not override user-provided fields for
+which it does not have its own opinion on ASO resources.
+
+Updating the embedded ASO API version in the CAPZ resources may not be possible to do safely without also
+bumping the CAPZ API version, however. Because ASO implements conversion webhook logic between several API
+versions for each AKS resource type, simply bumping the ASO API version in the CAPZ type without bumping the
+CAPZ API version would not allow that same conversion to be applied. This could lead to issues where a new
+version of CAPZ suddenly starts constructing invalid ASO resources and user intervention is required to
+perform the conversion manually.
+
+While it couples CAPZ to one ASO API version, this approach allows CAPZ to move a more calculated pace with
+regards to AKS API versions the way it's done today. This also narrows CAPZ's scope of responsibility which
+reduces CAPZ's exposure to potential incompatibilities with certain ASO API versions.
+
+Regarding ClusterClass, this option functions the same as [Option 2] or [Option 3], where all ASO fields can
+be defined in a template. This option opens up an additional safeguard though, where webhooks could flag
+fields which should not be defined in a template. Similar webhook checks would be less practical in the other
+options where CAPZ would need to be aware of the set of disallowed fields for each ASO API version that a user
+could use.
+
+Similarly, CAPZ's webhooks are also better able to validate and default the ASO configuration and ensure
+fields like ManagedCluster's `spec.owner` that should not be modified by users are set correctly.
+
+#### Option 5: No change: CAPZ resource evolution proceeds the way it currently does
+
+The final option proposed in this document is the option not to make any changes to how the CAPZ API is
+generally structured for AKS resources. CAPZ API types will continue to be curated manually without inheriting
+anything from the ASO API.
+
+Benefits of continuing on our current path include:
+- Familiarity with the existing pattern by users and contributors
+- Zero up-front cost to implement or transition to a new pattern
+- No requirement for users to have ASO knowledge
+- Greater freedom to change API implementations which we've recently leveraged to transition between the older
+  and newer Azure SDKs and to ASO.
+
+#### Decision
+
+[Option 5], we plan to stick with the current approach. All of the other options have significant drawbacks
+without any clear signs that any one of the sets of caveats is acceptable to carry long-term.
```

Comment:
Considering all of the current options have significant drawbacks, I'm advocating we pursue this new "none of the above" Option 5 and elect not to make any changes here. What would make me feel better about picking one of the other options is either an overwhelming amount of user consensus on one of the proposed options or a new option emerging which brings significantly fewer caveats than what's currently proposed.